From ed7f25819d5370aa16cb8fd0e99d4b79d6c0132c Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Wed, 8 Jul 2026 16:17:21 +0200
Subject: [PATCH 20/46] AnnaThermostat: add support for
 intended_central_heating_state

Parse intended_central_heating_state and publish it as an on/off switch sensor.

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
Assisted-by: GPT-5.3-Codex

Upstream-Status: Inappropriate [WIP]
---
 hardware/AnnaThermostat.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/hardware/AnnaThermostat.cpp b/hardware/AnnaThermostat.cpp
index 23a3e0660..6fd398866 100644
--- a/hardware/AnnaThermostat.cpp
+++ b/hardware/AnnaThermostat.cpp
@@ -1159,6 +1159,14 @@ void CAnnaThermostat::GetMeterDetails()
 					SendSwitch(applianceIDhex, 5, batterypercentage, isOn, 0, ApplianceName + " " + sname, m_Name);
 				}
 			}
+			else if (sname == "intended_central_heating_state")
+			{
+				if (!tmpstr.empty())
+				{
+					bool isOn = (strcmp(tmpstr.c_str(), "on") == 0);
+					SendSwitch(applianceIDhex, 6, batterypercentage, isOn, 0, ApplianceName + " " + sname, m_Name);
+				}
+			}
             else if (sname == "central_heater_water_pressure")
             {
                 if (!tmpstr.empty())
